backend/src/api/post-api.js

```diff
@@ -67,44 +67,79 @@ router.post('/login', async (req, res) => {
     const { email, password } = req.body;
     const auth = req.auth;
 
-    signInWithEmailAndPassword(auth, email, password).then((userCredential) => {
+    try {
+        const userCredential = await signInWithEmailAndPassword(auth, email, password);
         const user = userCredential.user;
         console.log("login email:", user.email);
-        const membership = Membership.findOne({ accountId: user.uid });
-        if(membership === null) {
-            return res.status(200).send({
-                status: 'success',
-                message: 'User logged in',
-                data: {
-                    accountId: user.uid,
-                    email: user.email,
-                    isOrg: false,
-                },
+        
+        const membership = await Membership.findOne({ accountId: user.uid });
+        
+        if (membership === null) {
+            console.log("member not found");
+            return res.status(404).send({
+                status: 'error',
+                message: 'Membership not found',
             });
```

Comment:
Because `membership` is being used to track if the `User` is a member of an `organization` or not. The user's membership can be `null` but we would still return `success` because it means that the member is a `volunteer`

Reply:
I see. I'll write a second query instead of trying to modify the first to work with that.